Makefile

```diff
@@ -1,4 +1,22 @@
-GO ?= $(shell command -v go 2> /dev/null)
+# Project variables
+NAME        := mattermost-mattermod
+AUTHOR      := mattermost
+URL         := https://github.com/$(AUTHOR)/$(NAME)
+
+# Build variables
+COMMIT_HASH  ?= $(shell git rev-parse --short HEAD 2>/dev/null)
+BUILD_DATE   ?= $(shell date +%FT%T%z)
+CUR_VERSION  ?= $(shell git describe --tags --exact-match 2>/dev/null || git describe --tags 2>/dev/null || echo "v0.0.0-$(COMMIT_HASH)")
+
+# Go variables
+GOLDFLAGS   :="
```

Comment:
I specifically didn't name it `LDFLAGS` to prevent environment variable collision that one might already set in their terminal and let `GOLDFLAGS` be dedicated to this Makefile targets. I'm completely fine with changing it though.

Reply:
My Makefile knowledge is very superficial, but I think this is just a Makefile variable and should not collide if an environment variable with the same name is set.